packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
     </TooltipHost>
   );
 };
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Find mention tag if selection is inside of it
+ *
+ * @private
+ */
+const findMentionTagForSelection = (tags: TagData[], selection: number): TagData | undefined => {
+  let mentionTag: TagData | undefined = undefined;
+  for (let i = 0; i < tags.length; i++) {
+    const tag = tags[i];
+    let plainTextEndIndex = 0;
+    if (tag.plainTextEndIndex !== undefined && tag.closeTagIdx !== undefined) {
+      // close tag exists
+      plainTextEndIndex = tag.plainTextEndIndex;
+    } else if (tag.plainTextBeginIndex !== undefined) {
+      //no close tag
+      plainTextEndIndex = tag.plainTextBeginIndex;
+    }
+    if (tag.subTags !== undefined && tag.subTags.length !== 0) {
+      const selectedTag = findMentionTagForSelection(tag.subTags, selection);
+      if (selectedTag !== undefined) {
+        mentionTag = selectedTag;
+        break;
+      }
+    } else if (
+      tag.tagType === 'msft-mention' &&
+      tag.plainTextBeginIndex !== undefined &&
+      tag.plainTextBeginIndex < selection &&
+      selection < plainTextEndIndex
+    ) {
+      mentionTag = tag;
+      break;
+    }
+  }
+  return mentionTag;
+};
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Go through tags and find a new the selection index if it is inside of a mention tag
+ *
+ * @private
+ */
+const findNewSelectionIndexForMention = (
+  tag: TagData,
+  textValue: string,
+  selection: number,
+  previousSelection: number
+): number => {
+  if (
+    tag.plainTextBeginIndex === undefined ||
+    tag.tagType !== 'msft-mention' ||
```

Comment:
Not fully understand this part - if not 'msft-mention' why do we still return selection?